hazelcast-tpc-engine/src/main/java/com/hazelcast/internal/tpc/iobuffer/ThreadLocalIOBuffer.java

```diff
@@ -0,0 +1,289 @@
+/*
+ * Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.tpc.iobuffer;
+
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+
+import static com.hazelcast.internal.tpc.iobuffer.ThreadLocalIOBufferAllocator.BUFFER_SIZE;
+import static com.hazelcast.internal.tpc.nio.Bits.INT_SIZE_IN_BYTES;
+import static com.hazelcast.internal.tpc.nio.Bits.LONG_SIZE_IN_BYTES;
+import static com.hazelcast.internal.tpc.nio.Bits.SHORT_SIZE_IN_BYTES;
+
+class ThreadLocalIOBuffer implements IOBuffer {
```

Comment:
I'm not crazy about the name ThreadLocalIOBuffer because it seems to be a ThreadLocal containing an IOBUffer.

Reply:
My understanding is simple : it is a io buffer was allocated by allocation which is local to single thread :)
Well, actually it's a good name for me, but suggestions are always welcome